### ISSUE TITLE
[postgres] Unparsable section for expressions in LIMIT or OFFSET

### ISSUE THREAD

**Issue by enote-kane** [sqlfluff/sqlfluff#4938]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Lint the following:
```sql
SELECT col_a
FROM test_table
LIMIT 10 * 10 OFFSET (5 * 100)
```

### Expected Behaviour

Passes

### Observed Behaviour

Raises an error:
```
L:   3 | P:   9 |  PRS | Line 3, Position 9: Found unparsable section: ' * 10
                       | OFFSET (5 * 100)'
```

### How to reproduce

Use this SQL:

```sql
SELECT col_a
FROM test_table
LIMIT 10 * 10 OFFSET (5 * 100)
```

### Dialect

postgres

### Version

2.1.1

### Configuration

_default_

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by joaostorrer** [comment]
This can be closed, since it was fixed and already merged
